libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+    // TODO: cxxLanguageStandard: CXXLanguageStandard? = nil
+
+    fun platforms(configure: SupportedPlatform.() -> Unit) {
+        val supportedPlatform = SupportedPlatform().apply(configure)
+        platforms = supportedPlatform.platforms
+    }
+
+    fun products(configure: Products.() -> Unit) {
+        val product = Products().apply(configure)
+
+        productsLibraries = product.libraries
+        productsExecutables = product.executables
+    }
+
+    fun dependencies(configure: Dependencies.() -> Unit) {
+        val dependency = Dependencies().apply(configure)
+        dependencies = dependency.dependencies
+    }
+
+    fun targets(configure: Targets.() -> Unit) {
+        val target = Targets().apply(configure)
+
+        targetsRegular = target.regulars
+        targetsExecutable = target.executables
+        targetsTest = target.tests
+    }
+
+    // TODO: add fun with another params (supportedPlatform)
+    class SupportedPlatform() {
+        var platforms = mutableListOf<SupportedPlatform>()
+
+        lateinit var type: PlatformType
+        lateinit var version: String
+
+        constructor(type: PlatformType, version: String) : this() {
+            this.type = type
+            this.version = version
+        }
+
+        fun macOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.macOS, version)
+            platforms.add(platform)
+        }
+
+        fun iOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.iOS, version)
+            platforms.add(platform)
+        }
+
+        fun tvOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.tvOS, version)
+            platforms.add(platform)
+        }
+
+        fun watchOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.watchOS, version)
+            platforms.add(platform)
+        }
+
+        enum class PlatformType {
+            macOS, iOS, tvOS, watchOS
+        }
+    }
+
+    class Products {
+        var libraries = mutableListOf<ProductType.Library>()
+        var executables = mutableListOf<ProductType.Executable>()
+
+        fun library(name: String, configure: ProductType.Library.() -> Unit) {
+            val library = ProductType.Library(name).apply(configure)
+            libraries.add(library)
+        }
+
+        fun library(name: String, configure: Closure<*>) = library(name) {
+            ConfigureUtil.configure(configure, this)
+        }
+
+        fun executable(name: String, configure: ProductType.Executable.() -> Unit) {
+            val executable = ProductType.Executable(name).apply(configure)
+            executables.add(executable)
+        }
+
+        fun executable(name: String, configure: Closure<*>) = executable(name) {
+            ConfigureUtil.configure(configure, this)
+        }
+
+        sealed class ProductType {
+            class Executable(@Input val name: String) {
```

Comment:
Nesting is not required here but significantly complicates code-readability. Make sources more flat, please.